zerver/models.py

```diff
@@ -64,10 +64,9 @@ def query_for_ids(query: QuerySet, user_ids: List[int], field: str) -> QuerySet:
     guard against empty lists of user_ids.
     '''
     assert(user_ids)
-    value_list = ', '.join(str(int(user_id)) for user_id in user_ids)
-    clause = '%s in (%s)' % (field, value_list)
+    clause = f'{field} IN %s'
     query = query.extra(
-        where=[clause]
+        where=[clause], params=(tuple(user_ids),),
     )
```

Comment:
This change is new. I benchmarked it as being about as fast as the previous code and still much faster than Django’s `filter(field__in=user_ids)`, presumably because the escaping is handled in psycopg2’s C rather than in Django’s Python. Maybe Django’s `__in` should be doing this when the DB is Postgres…